File: INTRODUCCION-POO/EJ_1/EJ_1.py
class Persona:
    def __init__(self, nombre, edad, ciudad):
        self.nombre = nombre
        self.edad = edad
        self.ciudad = ciudad

    def saludar(self):
        print(f"Hola, soy {self.nombre} de {self.ciudad}")

    def es_mayor_edad(self):
        return self.edad >= 18


persona1 = Persona("Nacho", 20, "La Paz")
persona2 = Persona("Lucía", 17, "Cochabamba")
persona3 = Persona("Mateo", 25, "Santa Cruz")

personas = [persona1, persona2, persona3]

for persona in personas:
    persona.saludar()
    if persona.es_mayor_edad():
        print("Es mayor de edad\n")
    else:
        print("Es menor de edad\n")

File: INTRODUCCION-POO/EJ_1/EJ_1.java
public class Persona {
    String nombre;
    int edad;
    String ciudad;

    public Persona(String nombre, int edad, String ciudad) {
        this.nombre = nombre;
        this.edad = edad;
        this.ciudad = ciudad;
    }

    public void saludar() {
        System.out.println("Hola, soy " + nombre + " de " + ciudad);
    }

    public boolean esMayorEdad() {
        return edad >= 18;
    }

    public static void main(String[] args) {
        Persona persona1 = new Persona("Nacho", 20, "La Paz");
        Persona persona2 = new Persona("Lucía", 17, "Cochabamba");
        Persona persona3 = new Persona("Mateo", 25, "Santa Cruz");

        Persona[] personas = {persona1, persona2, persona3};

        for (Persona p : personas) {
            p.saludar();
            if (p.esMayorEdad()) {
                System.out.println("Es mayor de edad\n");
            } else {
                System.out.println("Es menor de edad\n");
            }
        }
    }
}
